tests/kani/Intrinsics/bswap.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Check that `bswap` returns the expected results
+#![feature(core_intrinsics)]
+
+const BITS_PER_BYTE: usize = 8;
+
+macro_rules! test_bswap_intrinsic {
+    ($ty:ty, $check_name:ident, $get_byte_name:ident) => {
+        // Gets the i-th byte in `x`
+        fn $get_byte_name(x: $ty, i: usize) -> $ty {
+                let mask = 0xFF as $ty << i * BITS_PER_BYTE;
+                let masked_res = x & mask;
+                let bytes_res = masked_res >> i * BITS_PER_BYTE;
+                bytes_res
+        }
+
+        // Checks that the original byte is equal to the swapped byte
```

Comment:
Maybe change this to say checks that the order of the bytes in the original integer is reverse in the swapped one.